torch/_inductor/scheduler.py

```diff
@@ -681,6 +681,20 @@ def can_inplace(self, read_dep: dependencies.MemoryDep):
             return read_dep.index == write_dep.index and read_dep.size == write_dep.size
         return False
 
+    def has_atomic_add(self, check_buf):
+        # Return True if check_buf is stored by atomic_add mode
+        for node in self._body.get_nodes():
```

Comment:
is it better to build a cached helper method that returns the atomic_add buffer so that here we just check if `check_buf` equals to it? This way, you don't have to scan the body IR again and again.

Reply:
Thanks for the suggestion and changed. Yean, it's better and can save redundant body IR scan.